x/stakeibc/keeper/reward_converter.go

```diff
@@ -51,6 +53,40 @@ type ForwardMetadata struct {
 // and the normal staking and distribution flow will continue from there.
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// Builds an authz MsgGrant or MsgRevoke to grant an account trade capabilties on behalf of the trade ICA
+func (k Keeper) BuildTradeAuthzMsg(
+	ctx sdk.Context,
+	tradeRoute types.TradeRoute,
+	permissionChange types.AuthzPermissionChange,
+	grantee string,
+) (authzMsg []proto.Message, err error) {
+	swapMsgTypeUrl := "/" + proto.MessageName(&types.MsgSwapExactAmountIn{})
+
+	switch permissionChange {
+	case types.AuthzPermissionChange_GRANT:
+		authorization := authz.NewGenericAuthorization(swapMsgTypeUrl)
```

Comment:
nit: is it possible to limit the authorization to only swap from token A to token B? imo this would give a small amount of additional protection / make the system easier to reason about, but not critical if it can't be done!

Reply:
it's a good question! I'm not sure. @riley-stride do you know?

Reply:
Got it, good to know. We should mark this as a fast follow, it doesn't seem terribly hard to do, and it sounds like the Osmosis team is conceptually on-board.